tests/integration/acp.go

```diff
@@ -310,11 +310,192 @@ func addDocActorRelationshipACP(
+	DocID int
+
+	// The name of the relation within the relationship we want to delete (should be defined in the policy).
+	//
+	// This is a required field.
+	Relation string
+
+	// The target public identity, i.e. the identity of the actor with whom the relationship is with.
+	//
+	// This is a required field. To test the invalid usage of not having this arg, use -1 index.
+	TargetIdentity int
+
+	// The requestor identity, i.e. identity of the actor deleting the relationship.
+	// Note: This identity must either own or have managing access defined in the policy.
+	//
+	// This is a required field. To test the invalid usage of not having this arg, use -1 index.
+	RequestorIdentity int
+
+	// Result returns true if the relationship record was expected to be found and deleted,
+	// and returns false if no matching relationship record was found (no-op).
+	ExpectedRecordFound bool
+
+	// Any error expected from the action. Optional.
+	//
+	// String can be a partial, and the test will pass if an error is returned that
+	// contains this string.
+	ExpectedError string
+}
+
+func deleteDocActorRelationshipACP(
+	s *state,
+	action DeleteDocActorRelationship,
+) {
+	if action.NodeID.HasValue() {
+		nodeID := action.NodeID.Value()
+		collections := s.collections[nodeID]
+		node := s.nodes[nodeID]
+
+		var collectionName string
+		if action.CollectionID == -1 {
+			collectionName = ""
+		} else {
+			collection := collections[action.CollectionID]
+			if !collection.Description().Name.HasValue() {
+				require.Fail(s.t, "Expected non-empty collection name, but it was empty.", s.testCase.Description)
+			}
+			collectionName = collection.Description().Name.Value()
+		}
+
+		var docID string
+		if action.DocID == -1 || action.CollectionID == -1 {
+			docID = ""
+		} else {
+			docID = s.docIDs[action.CollectionID][action.DocID].String()
+		}
+
+		var targetIdentity string
+		if action.TargetIdentity == -1 {
+			targetIdentity = ""
+		} else {
+			optionalTargetIdentity := getIdentity(s, nodeID, immutable.Some(action.TargetIdentity))
+			if !optionalTargetIdentity.HasValue() {
+				require.Fail(s.t, "Expected non-empty target identity, but it was empty.", s.testCase.Description)
+			}
+			targetIdentity = optionalTargetIdentity.Value().DID
+		}
+
+		var requestorIdentity immutable.Option[acpIdentity.Identity]
+		if action.RequestorIdentity == -1 {
+			requestorIdentity = acpIdentity.None
+		} else {
+			requestorIdentity = getIdentity(s, nodeID, immutable.Some(action.RequestorIdentity))
+			if !requestorIdentity.HasValue() {
+				require.Fail(s.t, "Expected non-empty requestor identity, but it was empty.", s.testCase.Description)
+			}
+		}
+		ctx := db.SetContextIdentity(s.ctx, requestorIdentity)
+
+		deleteDocActorRelationshipResult, err := node.DeleteDocActorRelationship(
+			ctx,
+			collectionName,
+			docID,
+			action.Relation,
+			targetIdentity,
+		)
+
+		if err == nil {
```

Comment:
question: Why are these requires wrapped in an if, and not just done after the calls to `AssertError` and `assertExpectedErrorRaised`?

Reply:
Is copy pasted from `addPolicy` and the `addDocActorRelationshipACP`, I don't remember of the top of my head.

Reply:
Can you change both then please?

Reply:
> Can you change both then please?

Sure, I don't mind :)

Reply:
- [ ] Remove `err == nil` if check from test

Reply:
Update: I did remove the asserts in the end and the tests started failing it's because it does the checks weather an error was raised or not. I am undoing the change. If I can remember I did do this change because It would upon failure sometimes show a false positive error not raised. I am too lazy to understand why haha but the current framework works, so keeping as is

Reply:
I would guess that you forgot to return/continue early if `expectedErrorRaised == true` and continued to the value assertion after receiving the expected error

Reply:
But then that's still another if check anyways. I am happy with the current setup. Is fairly easy to change later for all the test actions if we want.

Reply:
But then that's still another if check anyways. I am happy with the current setup. Is fairly easy to change later for all the test actions if we want.

Reply:
Another `if` yes, but you are no longer spoofing the main error assertion logic.

Reply:
Resolved in commit: ``` PR(ADHERE-REVIEW): Check `require.Equal` after the `assertExpectedErrorRaised` ```